openapi.yaml

```diff
@@ -164,1090 +135,1589 @@ paths:
+        example: '1970-01-01'
+  AddressResource:
+    type: object
     properties:
-      data:
-        properties:
-          type:
-            $ref: '#/definitions/SSNType'
-          attributes:
-            required:
-              - ssn
-            properties:
-              ssn:
-                type: string
-                pattern: '\d{9}'
-  SSNResultObject:
+      id:
+        type: string
+        description: Globally unique ID for the address
+      type:
+        $ref: '#/definitions/AddressType'
+      links:
+        $ref: '#/definitions/SelfLink'
+      attributes:
+        allOf:
+          - $ref: '#/definitions/AddressPostAttributes'
+          - $ref: '#/definitions/AddressAttributes'
+  AddressSetResult:
+    type: object
     properties:
       links:
         $ref: '#/definitions/SelfLink'
       data:
+        type: array
+        items:  
+          $ref: '#/definitions/AddressResource'
+  AddressResult:
+    type: object
+    properties:
+      links:  
+        $ref: '#/definitions/SelfLink'
+      data:
+        $ref: '#/definitions/AddressResource'
+  AddressPostBody:
+    type: object
+    required:
+      - data
+    additionalProperties: false
+    properties:
+      data:
+        type: object
+        required:
+          - type
+          - attributes
+        additionalProperties: false
         properties:
-          id:
-            type: string
-            description: 9 digits SSN
-            example: '123456789'
           type:
-            $ref: '#/definitions/SSNType'
-          links:
-            $ref: '#/definitions/SelfLink'
+            $ref: '#/definitions/AddressType'
           attributes:
-            properties:
-              ssnStatus:
-                type: string
-                description: SSN status
-                enum:
-                  - vault
-                  - valid
-                  - invalid
-  NewPhoneResultObject:
+            allOf:
+              - $ref: '#/definitions/AddressPostRequired'
+              - $ref: '#/definitions/AddressPostAttributes'
+  PhonePostRequired:
+    type: object
+    required:
+      - addressType
+      - phoneType
+      - areaCode
+      - phoneNumber
+      - primaryInd
+  PhonePostAttributes:
+    type: object
+    additionalProperties: false
     properties:
-      data:
-        $ref: '#/definitions/NewPhoneResourceObject'
-  NewPhoneResourceObject:
+      areaCode:
+        type: string
+        description: Area code for the phone number.
+        example: '541'
+        pattern: '^[0-9]{1,3}$'
+      phoneNumber:
+        type: string
+        description: Extension for the phone number. Can be returned in different formats.
+        example: '4443333'
+        pattern: '^[0-9]{1,7}$'
+      phoneExtension:
+        type: string
+        description: Extension for the phone number. Can be returned in different formats.
+        example: '20'
+        pattern: '^[0-9]{1,4}$'
+      primaryInd:
+        type: b
```

Comment:
It looks like these email post attributes are only used as part of the GET /.../emails endpoint. If there won't be a POST endpoint, could you just move these into EmailAttributes?

Reply:
Good catch. There was a post for email but it got removed.